mock/tests/testasync.py

```diff
@@ -0,0 +1,636 @@
+
+import asyncio
+import inspect
+import unittest
+
+from mock import call, AsyncMock, patch, MagicMock, create_autospec
+from mock.mock import _AwaitEvent
+
+
+try:
+    from asyncio import run
+except ImportError:
+    def run(main):
```

Comment:
Returning the value will help in fixing some of the cases : 

```python
def run(main):
    loop = asyncio.new_event_loop()
    return_value = loop.run_until_complete(main)
    loop.close()
    return return_value
```

Reply:
yep, that got them all!